### DISCUSSION TITLE
Real-time monitoring

### DISCUSSION THREAD

**Discussion by Ioan-Popa** [sidorares/node-mysql2#2304] [General]
Hello everybody,

does this package offer real-time monitoring capabilities of a MySQL server? I wish to get notifications in a Node.js application whenever an UPDATE or INSERT happens in a certain table. These changes should come from anywhere, not only from the Node.js application itself.

If this package is not suited for this purpose, would you be able to point me to one that is?

Thank you.

**Comment by sidorares** [comment]
you could use replication / binlog client for that. I haven't tried this in a long time so not sure about current status. Unfortunately there is not much documentation. If you want to experiment with it, read a tutorial on how to setup replication in mysql server and then try to connect with  `createBinlogStream` https://github.com/sidorares/node-mysql2/blob/489154f98224fad8036b87c63df8b736ede1c4aa/lib/connection.js#L771

- you'll get stream of data and schema updates in a datbase

**Reply by Ioan-Popa** [reply]
Thank you. I figured out how to perform real-time monitoring with the help of the @rodrigogs/mysql-events package.
